src/components/override-sdk/infra/Assignment/Assignment.tsx

```diff
@@ -310,6 +310,18 @@ export default function Assignment(props) {
     }
   }, [actionButtons]);
 
+  // Added a beforeunload event to save an existing claim if the page is refreshed
+  useEffect(() => {
+    const handleBeforeUnload = () => {
+      buttonPress('saveAssignment', 'secondary');
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, []);
+
```

Comment:
This is duplicate logic as same is handled in DefaultForm.tsx. One of these need to be removed.